lib/portage/package/ebuild/fetch.py

```diff
@@ -408,23 +408,25 @@ def get_mirror_url(mirror_url, filename, cache_path=None):
 	if ts >= time.time() - 86400:
 		mirror_conf.deserialize(data)
 	else:
+		tmpfile = '.layout.conf.%d' % time.time()
```

Comment:
We really need to use mkstemp to solve races here.

Reply:
And for FEATURES=skiprocheck read-only fetching it's good enough to use the existing urlopen code.

Reply:
In practice, we could probably get away with generating a random string instead of using mkstemp (and a random string would work for FEATURES=skiprocheck). But technically, mkstemp is the only truly reliable approach.

Reply:
Using a constant path containing the mirror name could work if we add a fetch parameter to force fetch even when the file exists already. We'd simply leave the file in DISTDIR, and stipulate that FEATURES=skiprocheck users should have a FETCHCOMMAND that replaces the file atomically.

Reply:
I've rewritten it to use `NamedTemporaryFile` but having the file exist makes Portage rename it to `_checksum_failure_` blah blah. Which of course reintroduces the race.

Reply:
I see `repoman` is also written with the brilliant idea of using `mkstemp()` and immediately discarding the file, rendering `mkstemp()` totally useless.

Reply:
Let's use a constant path containing the mirror name, assume that fetch updates it atomically, and don't remove it from DISTDIR. This approach will work with FEATURES=skiprocheck.

Reply:
Do you have any particular method of transforming mirror URI into safe filename component in mind?

Reply:
I imagined that we could use the hostname (fqdn), since it doesn't seem like there's a reason to have multiple mirrors per hostname.

Reply:
WFM.